### ISSUE TITLE
Question: License

### ISSUE THREAD

**Issue by tdbgamer** [bodil/im-rs#10]
This is a very exciting library you've made here. Looking at bit at the MPL license seems like it would be nearly impossible for many people to use this for work. I can't speak for everyone, but at least at my job we can basically only use the more permissive BSD/MIT/Apache licenses. Most of the Rust ecosystem uses a combination of MIT and Apache which makes it a lot easier to use in a work environment.

I'm not trying to pressure you to change it or anything, but I'm just curious if there's a particular reason you chose MPL and if you had any attachment to it.

Either way, great work on this library!

**Comment by bodil** [comment]
Short answer: yes, the licence is very deliberate and not likely to change, especially to BSD et al.

What is it about the licence that you think would make it impossible for many people to use? If I've missed something particularly egregious, which I must have if your reading is correct, I'd be willing to consider relicensing, but it really does sound like you may have misunderstood the scope of the licence.

**Comment by tdbgamer** [comment]
Definitely possible I'm misunderstanding it.

Main reason I was a little worried is because of the sections on distribution. It seemed very similar to GPL and most place I've worked have a blanket rule to disallow all use of GPL just in case that section could be interpreted as having to reveal all source code (even if that wasn't the original intent of the license). It's usually viewed as being too risky since you don't know how it will be interpreted until you go to court and they end up either rewriting the library or finding an alternative with a different license. Most of the Rust ecosystem (even the Rust compiler) uses the MIT/Apache combination and those are generally very business friendly and considered low risk, but I'm sure you have your reasons for choosing MPL.

Maybe I'm way off and just being overly cautious though.

**Comment by bodil** [comment]
Yeah, you're actually touching on the reason I changed it from LGPL to MPL - that irrational fear of the "viral" licence. The big difference between the MPL and the LGPL is that the MPL is designed so it applies only to the individual source files its assignment header has been added to, thus providing a very clear legal boundary. So, if you change any of the files in this package, you're required to publish those changes in a manner similar to the GPL's, but it has absolutely no impact on any of the code outside of this package, or even the files in this package which don't contain the MPL header (I think the `Cargo.toml` file might have inadvertently escaped the relicensing). It's explicitly non-"viral" in scope.

Although, IANAL etc and various MPL assignment texts around the project might mean it applies to the whole thing, but it still very deliberately only applies to this project, not any projects using it.

As to why MPL over BSD and friends: I don't want to be business friendly as much as I want to ensure business is friendly in return. I'm OK with this code being used in proprietary software (hence no GPL) but I also want to ensure that if someone fixes a bug in my code, or figures out how to make it go faster, they contribute that knowledge back to the community (hence no BSD).

Corporate lawyers are usually OK with the MPL - if they've got a list and it's not on it, I'd ask them to review it and reconsider. On the flip side, the MPL is also compatible with just about any other open source licence out of the box, GPL included.

Anyway, closing. 😊
